Extend auction interaction feat test

witgaw opened the pull request:
Extends and modifies the auction interaction feature test + implements functionality required to make it pass.

The main issue is that the auction trigger field in market data API doesn't always get modified to reflect the trigger that has extended the auction (https://github.com/vegaprotocol/specs-internal/blob/master/protocol/0026-auctions.md#exiting-the-auction-mode).

This PR also brings implementation in line with spec changes, please see linked issues for details.

Blocked by: ~https://github.com/vegaprotocol/vega/issues/3570 https://github.com/vegaprotocol/vega/issues/3578~ ~https://github.com/vegaprotocol/vega/pull/5298~

Closes: https://github.com/vegaprotocol/vega/issues/3047 https://github.com/vegaprotocol/vega/issues/3570 https://github.com/vegaprotocol/vega/issues/4223 https://github.com/vegaprotocol/vega/issues/3362 https://github.com/vegaprotocol/vega/issues/5237

witgaw committed:
test: extend auction interaction test (TODOs in place)

+ initial code changes (still WIP)

witgaw committed:
feat: use indicative ref price in auction

witgaw committed:
feat: all scenarios in 0026-AUCT... passing

Apply code changes so that all tests in 0026-AUCT... pass (test has been
modified to reflect current specs)

witgaw committed:
test: fix unit tests

witgaw committed:
test: fix integration tests

witgaw committed:
ci: fix linter errors

witgaw committed:
style: address gofumpt errors

witgaw committed:
ci: gofumpt files

witgaw committed:
refactor: simplify execution/auction

witgaw committed:
refactor: uncomment code

witgaw committed:
chore: revert launch.json to previous state

witgaw committed:
chore: remove reduntant TODOs

witgaw committed:
ci: update CHANGELOG

witgaw committed:
ci: update CHANGELOG

witgaw committed:
ci: gofumpt much

witgaw committed:
style: remove redundant else

witgaw committed:
ci: lint

witgaw committed:
refactor: remove GetIndicativeTrades

witgaw committed:
style: use consistent  lp type

witgaw committed:
refactor: clarify getReferencePrice

witgaw committed:
refactor: simplify logic in checkLiquidity

witgaw committed:
refactor: move common steps before branching

witgaw committed:
refactor: expose and use PM internal variable

witgaw committed:
refactor: remove redundant variable

witgaw committed:
ci: comment should end in a period

EVODelavega reviewed (changes requested):
As pointed out in individual comments, and summarised on slack, I think we require some changes to the code still. The most important things I've noticed are:

* With the current implementation, I believe it's possible for someone to spam FOK/IOC orders at stupid prices to force a price auction to get triggered
* `ExtractOrders` can now return pointers to orders as they sit on the book, making it very easy to introduce bugs like: having an order at price level 1000, but reprice it to 2000 without updating the matching engine
* Possible changes to auction behaviours (comment on integration tests about that)
* Auction extensions effectively being treated as leaving 1 type of auction and starting a second auction immediately after (which as per spec should never happen)
* Snapshots may be broken
* If we have a horizon of, say 10s, and we get an indicative price/volume for the first time in the last seconds of the opening auction phase, the price bounds may be off, and we might trigger a panic when the opening auction ends. I think we should add an integration test that replicates this scenario to ensure this isn't the case

witgaw committed:
style: revert formatting changes

witgaw committed:
refactor: remove redundant call

witgaw committed:
refactor: simplify loop

witgaw committed:
refactor: return orders as values

witgaw committed:
refactor: clone orders

witgaw committed:
refactor: split auction and extension helper funcs

witgaw committed:
fix: regenerate mocks

witgaw committed:
refactor: don't return errors from monitors

witgaw committed:
ci: udpate CHANGELOG

witgaw committed:
fix: post rebase problems

witgaw committed:
fix: amend integration tests following rebase

witgaw committed:
refactor: allow optional check for reference price

witgaw committed:
style: improve ref price readability

witgaw committed:
refactor: don't skip blocks, adjust to bounds

witgaw reviewed (commented):
Addressed most of the comments

witgaw committed:
refactor: remove redundant loop

witgaw committed:
refactor: advance time 2x 150 blocks

witgaw committed:
refactor: revert to last price

Revert to previous behaviour where we reset price history to last price
The really is incorrect, but a proper fix to
this should be done in a separate PR.

witgaw committed:
fix: modify test to use old triggering ratio

witgaw committed:
fix: modify test so that we leave auction

witgaw committed:
fix: only check liquidity out of auction

There's no need to check reference moves and liquidity when
market is in auciton mode, this will be done upon trying to leave it.

witgaw committed:
ci: remove whitespace

witgaw committed:
test: first vs final uncrossing price

witgaw committed:
refactor: write for loop properly
